pkgs/development/python-modules/sanic/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, buildPythonPackage, fetchPypi, pythonOlder
+, aiofiles, websockets, multidict, ujson, uvloop, httptools
+}:
+
+let
+  # Ideally, websockets 7.0 will be supported in the future, but
+  # for now, sanic requires 6.x
+  ws = websockets.overrideAttrs (_: {
```

Comment:
We can't have a different version of a package in `python-packages.nix` as that will result in collisions.

Reply:
Am I missing something here, or are you suggesting that `python-packages.nix` infrastructure cannot feature any version overrides in any expressions? That seems... extremely not-Nix like. Or are you worried that `ws` will end up pulled in by possible downstream users of `sanic` (due to `propagatedBuildInputs`)?

Reply:
Python cannot handle having multiple versions of a module in a single closure. If someone wants to build a Python env with this package, and another package that uses the other version of `ws`, they will get a collision. Now, one could say that is a risk/problem for the user whenever they compose such environment, but that would severely reduce the quality of the package set.